sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1243,6 +1249,24 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
         IsNotNull(e)
       case SqlBaseParser.NULL =>
         IsNull(e)
+      case SqlBaseParser.TRUE if ctx.NOT != null =>
+        checkBooleanTestArgs(e)
+        Not(BooleanTest(e, Some(true)))
+      case SqlBaseParser.TRUE =>
+        checkBooleanTestArgs(e)
+        BooleanTest(e, Some(true))
+      case SqlBaseParser.FALSE if ctx.NOT != null =>
+        checkBooleanTestArgs(e)
+        Not(BooleanTest(e, Some(false)))
+      case SqlBaseParser.FALSE =>
+        checkBooleanTestArgs(e)
+        BooleanTest(e, Some(false))
+      case SqlBaseParser.UNKNOWN if ctx.NOT != null =>
+        checkBooleanTestArgs(e)
+        Not(BooleanTest(e, None))
```

Comment:
Ur, IIRC, @maropu 's suggestion is the following. @maropu . Could you check this again?
```scala
       case SqlBaseParser.UNKNOWN if ctx.NOT != null =>
         checkBooleanTestArgs(e)
-        Not(BooleanTest(e, None))
+        IsNotNull(e)
       case SqlBaseParser.UNKNOWN =>
         checkBooleanTestArgs(e)
-        BooleanTest(e, None)
+        IsNull(e)
```

Reply:
Ping, @beliefer .

Reply:
Yea, yes!

Reply:
Also, could you please add tests in `SQLQueryTestSuite` to check if the parser works well?

Reply:
@dongjoon-hyun OK. I changed it.

Reply:
OK. I will add test in `ExpressionParserSuite` to check parser.

Reply:
@dongjoon-hyun I have moved `checkBooleanTestArgs` from `AstBuilder` into `BooleanTest`. Because there can't know the fact data type when parsing it, until running time.

Reply:
I made a pr to your github based on your impl. Could you check there? https://github.com/beliefer/spark/pull/1

Reply:
@maropu I have checked your pr. Thanks for your help.
I will use `ExpectsInputTypes` to replace `checkBooleanTestArgs`.
I have also considered implementing `IsTrue`, `IsFalse`, `IsUnknown` separately, like `isNull`.
But I think `BooleanTest` is qualified for this job and reduce code line.

Reply:
If so, how about creating a shared abstract class for `IsTrue`, `IsNotTrue`, `IsFalse`, and `IsNotFalse`; https://github.com/beliefer/spark/pull/1/files#diff-aec2bffff270efab6835320c7a1dc8e9R845

Reply:
@maropu This is a better idea. I will reference your suggestion.